tools/proto-compiler/src/cmd/compile.rs

```diff
@@ -116,15 +121,11 @@ impl CompileCmd {
         // List available paths for dependencies
         let includes: Vec<PathBuf> = proto_includes_paths.iter().map(PathBuf::from).collect();
 
-        let attrs_serde =
-            r#"#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]"#;
-        let attrs_serde_default = r#"#[cfg_attr(feature = "serde", serde(default))]"#;
```

Comment:
This `serde(default)` attribute derivation should be preserved as well. One notable use case: this enables ibc-rs to keep supporting chains on different proto versions where token transfer packet data may come with or without `memo` field, so [this test ](https://github.com/cosmos/ibc-rs/blob/8c1766fa823f8a2e8e83a79f39c521aa6e270a81/ibc-apps/ics20-transfer/types/src/packet.rs#L110-L115) shouldn't fail by our changes here.
To address this, I've tried `pbjson` builder with calling the `emit_fields()` too which worked and resulted in the same behaviour as `attrs_serde_default`. Though this `emit_fields()` acts on all the proto types then. Please check this out on your end as well to ensure we won't be ending up with unexpected behaviours.

Reply:
If I understand this correctly, when serializing the struct, we want the json string to contain the default fields instead of omitting them entirely. Just added that compile option and two tests to ensure it's retained in the future.

Reply:
Can you confirm this test case matches the use-case you mentioned? https://github.com/cosmos/ibc-proto-rs/pull/166/files#diff-e7b72ba7bfc4f255bd7b4008c8c6aac51fd57a8db6869f242eef9274be74b12fR15-R23

Reply:
Exactly. You got it. 
By the way, I've added an additional test for the particular case of `TokenTransferPacketData`. So if we ever revisit the proto-compiler down the road, we'll get a heads-up on that.

Thank you @grod220 for addressing the comments! All set now from my end!